runtime/backtrace.c

```diff
@@ -349,5 +349,11 @@ CAMLprim value caml_get_exception_backtrace(value unit)
 }
 
 CAMLprim value caml_get_current_callstack(value max_frames_value) {
-  return caml_get_current_callstack_impl(Long_val(max_frames_value), 0);
+  CAMLparam1(max_frames_value);
+  CAMLlocal1(res);
+
+  res = caml_alloc(caml_current_callstack_size(Long_val(max_frames_value)), 0);
+  caml_current_callstack_write(res);
```

Comment:
Compared to the previous version, you do not return an atom anymore when the size is 0 -- you are doing this in your own `capture_callstack` function. I don't know if that is correct -- will `caml_alloc` work correctly if the size is 0?

Reply:
Yes, this works fine. `caml_alloc` returns `Atom(tag)` when the size is zero. I'm not sure why this check was duplicated in the original backtrace code.

Reply:
I see; the logic in `capture_callstack` is necessary because `caml_alloc_shr_aux` will not do this. Thanks for the clarification!

Reply:
Ah, of course, it's still needed in the alloc_shr case.